### ISSUE TITLE
Installing an invalid version displays an error, but installs the latest release anyways

### ISSUE THREAD

**Issue by RuyiLi** [taylor/kiex#99]
Specifying an invalid version in the install command (e.g. `kiex install 17.0`) initially displays the following error messages:
```
fatal: 'v17.0' is not a commit and a branch 'v17.0' cannot be created from it
error: pathspec 'v17.0' did not match any file(s) known to git
fatal: couldn't find remote ref v17.0
```
but then proceeds to install what I assume is the latest version in the Elixir repository (i.e. whatever's specified in [the VERSION file on the main branch](https://github.com/elixir-lang/elixir/blob/main/VERSION)) under the name of the invalid version. This leads to some unexpected behavior like
```
$ kiex use 17.0-
$ elixir -v
...
Elixir 1.18.0-dev (64e4b89) (compiled with Erlang/OTP 27)
```

**Comment by taylor** [comment]
Good catch @RuyiLi.  Please show the .kiex folder with the different versions (using ls -l).  

This line
https://github.com/taylor/kiex/blob/main/kiex#L297 from 
```  if [ "$1" = "use" ] ; then
    source_line="$(USERSHELL=notcsh $HOME/.kiex/bin/kiex $*|grep source)"
    if [ -z "$source_line" ] ; then
      echo "Failed to $*"
      return 1
    else
```
should ensure it does not continue, but *-z* is looking for an empty string, so something is coming back and it continues... even though what is coming back is probably causing a problem.

@RuyiLi want to try fixing and submit a PR?

**Comment by RuyiLi** [comment]
Sure, I'll give it a shot once I'm free :)

```
$ ls -l ~/.kiex/elixirs/
total 8
drwxr-xr-x 5 ruyili ruyili 4096 Dec  7 20:17 elixir-1.17.3-
-rw-r--r-- 1 ruyili ruyili  153 Dec  7 20:17 elixir-1.17.3-.env
```

**Comment by taylor** [comment]
@RuyiLi still using kiex? Would you like to submit that PR? :)

**Comment by RuyiLi** [comment]
Oops, thanks for reminding me; #105
